packages/block-library/src/rss/index.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * Server-side rendering of the `core/rss` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Renders the rss error message.
+ *
+ * @param string $message RSS error message.
+ * @return string|void Void if renders on frontend.
+ */
+function render_rss_error_message( $message ) {
+	return '<div class="components-placeholder"><div class="notice notice-alt notice-error">' . $message . '</div></div>';
+}
+
+/**
+ * Renders the `core/rss` block on server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string Returns the block content with received rss items.
+ */
+function render_block_core_rss( $attributes ) {
+
+	$url = $attributes['feedURL'];
+
+	if ( empty( $url ) ) {
+		return;
+	}
+
+	while ( stristr( $url, 'http' ) !== $url ) {
+		$url = substr( $url, 1 );
+	}
+
+	// self-url destruction sequence.
+	if ( in_array( untrailingslashit( $url ), array( site_url(), home_url() ) ) ) {
+		return render_rss_error_message( __( 'Use \'Latest Posts\' block for this domain', 'gutenberg' ) );
+	}
+
+	$rss = fetch_feed( $url );
+
+	if ( is_wp_error( $rss ) ) {
+		return render_rss_error_message( '<strong>' . __( 'RSS Error:', 'gutenberg' ) . '</strong> ' . $rss->get_error_message() );
```

Comment:
Looking at screenshots in the description, I started wondering whether we should hide that very developer oriented messages from the users or at least warn about its technical nature :)